lib/beaker/host/unix/exec.rb

```diff
@@ -12,16 +12,17 @@ module Unix::Exec
   #
   # Will throw an exception RebootFailure if it fails
   def reboot(wait_time=10, max_connection_tries=9, uptime_retries=18)
+    require 'time'
+
     begin
-      original_uptime = exec(Beaker::Command.new("uptime"))
-      original_uptime_str = parse_uptime original_uptime.stdout
-      original_uptime_int = uptime_int original_uptime_str
+      original_boot_time_str = exec(Beaker::Command.new('who -b'), {:max_connection_tries => max_connection_tries, :silent => true}).stdout
+      original_boot_time_line = original_boot_time_str.lines.grep(/boot/).first
 
-      if self['platform'] =~ /solaris/
-        exec(Beaker::Command.new("reboot"), :expect_connection_failure => true)
-      else
-        exec(Beaker::Command.new("/sbin/shutdown -r now"), :expect_connection_failure => true)
-      end
+      raise Beaker::Host::RebootFailure, "Could not find system boot time using 'who -b': #{original_boot_time_str}" unless original_boot_time_line
+
+      original_boot_time = Time.parse(original_boot_time_line)
+
+      exec(Beaker::Command.new('/bin/systemctl reboot -i || reboot || /sbin/shutdown -r now'), :expect_connection_failure => true)
```

Comment:
Out of curiousity which system didnt take `/sbin/shutdown -r now` but would accept `/bin/systemctl reboot` ?
We could also add 
```suggestion
      exec(Beaker::Command.new('/bin/systemctl reboot -i || reboot || /sbin/shutdown -r now || init 6'), :expect_connection_failure => true)
```

Reply:
@sbeaulie EL7+.

`shutdown -r now` only works if no other user/daemon/whatever is actively logged in or otherwise blocking.

Since we're testing....whatever....there's really no way to guarantee that one of the offending activities is not occurring.